### DISCUSSION TITLE
DayContent missing from CustomComponents

### DISCUSSION THREAD

**Discussion by hrypkema-amplify** [gpbl/react-day-picker#2406] [Support] [answered]
### Issue

When configuring the DayPicker without a `mode`, the `DayContent` component is not customizable in the the same way as customizing the `DayButton` as its missing from `CustomComponents`

### Code

List of custom components: https://github.com/gpbl/react-day-picker/tree/main/src/components

Types: 
https://github.com/gpbl/react-day-picker/blob/095d40ca6849c6f8d27ecb4dfa603840c86fe1de/src/types/shared.ts#L47-L92

It is also referenced in the `DayButton` component:

https://github.com/gpbl/react-day-picker/blob/095d40ca6849c6f8d27ecb4dfa603840c86fe1de/src/components/DayButton.tsx#L9

### Question

Can `DayContent` be added as a component to be customized?

If not, is there an example how to set up a custom `Day` component to handle rendering either a custom `DayContent` or `DayButton` component based on `mode`?

**Comment by gpbl** [comment, marked as answer]
@hrypkema-amplify Yes, the `DayContent` component has been removed in v9. 

If you want to customize the content of the day when is not the `DayButton` component, pass a custom [formatDay](https://daypicker.dev/api/functions/formatDay) formatter:

```tsx
<DayPicker
  formatters={{
    formatDay: (day) => String(day.getDate())
  }}
/>
```

Thanks for reporting that comment in the documentation. I'm going to remove it...

**Comment by hrypkema-amplify** [comment]
@gpbl ah I see, thanks!

the use case I needed was rather to add a wrapping span element around `DayContent` only, and not `DayButton`.

in case its useful, the solution I ended up with was to create a custom `Day` component:

```tsx
const CustomDay = ({ day, modifiers, children, mode,  ...props }) => {
  const isInteractive = mode !== undefined;

  return (
    <td {...props}>
      {!isInteractive && (
        <span className="rdp-day_content">
          {children}
        </span>
      )}
      {isInteractive && children}
    </td>
  );
};
```

And customizing the DayPicker with it:

```tsx
import CustomDay from './CustomDay.tsx';

...

return (
  <DayPicker
    components={{
      Day: props => CustomDay({ mode, ...props }),
    }}
  />
);
```
